### PR TITLE
add resolved schema tool

### PR DESCRIPTION
**Pull request by zemberdotnet**
# Description

This PR adds `generate-resolved-schemas` to the repo.

This tool fully resolves each Schema with it's parent's properties.
Parent's properties are marked as `inherited`. It produces a single file for external
consumers.

### CODE REVIEW

**Comment by zemberdotnet** [comment]
`external/resolvedSchemas` ends up being ~750KBs if formatted with prettier ~566KBs if formated without it.

File: tools/generate-resolved-schemas.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    const filePath = path.join(dir, file);
+    const schemaBlob = await fs.readFile(filePath);
+    const parsedSchema = JSON.parse(schemaBlob.toString());
+    schemas.push(parsedSchema);
+  }
+
+  return schemas;
+}
+
+void (async function () {
```
Comment (VDubber) [inline tools/generate-resolved-schemas.js:142]: Would a test be valuable?

**Review by VDubber — COMMENTED** [review]
This is cool!
A few thoughts:
1. resolvedSchema is not added to the npm package, correct?
2. Should we add the new generate command to husky so it stays up-to-date?
3. Unit tests?

**Comment by zemberdotnet** [comment]
> 1. resolvedSchema is not added to the npm package, correct?
Correct!
> 2. Should we add the new generate command to husky so it stays up-to-date?
Yes, we should great call out.
> 3. Unit tests?
Yep. Let me see what it'd take to add tests for it.

File: tools/generate-resolved-schemas/utilt.test.js [hunk 2]
```diff
@@ -0,0 +1,509 @@
+const { resolveSchema, buildIdToSchemaMap } = require('./utils');
```
Comment (VDubber) [inline tools/generate-resolved-schemas/utilt.test.js:1]: nit: rename file to `utils` instead of `utilt`

**Review by VDubber — APPROVED** [review]
Awesome!

File: tools/generate-resolved-schemas/utilt.test.js (hunk 2 above)
Comment (zemberdotnet) [reply tools/generate-resolved-schemas/utilt.test.js:1]: 🤦 I'll fix that haha

**Review by VDubber — APPROVED** [review]

**Pull request merged**